specification/ml/_types/DiscoveryNode.ts

```diff
@@ -17,14 +17,32 @@
  * under the License.
  */
 
-import { Dictionary } from '@spec_utils/Dictionary'
-import { Id, Name } from '@_types/common'
+import { Dictionary, SingleKeyDictionary } from '@spec_utils/Dictionary'
+import { Id, Name, VersionString } from '@_types/common'
 import { TransportAddress } from '@_types/Networking'
+import { integer } from '@_types/Numeric'
 
-export class DiscoveryNode {
-  attributes: Dictionary<string, string>
+export type DiscoveryNode = SingleKeyDictionary<Id, DiscoveryNodeContent>
+
+export class DiscoveryNodeContent {
+  name?: Name
   ephemeral_id: Id
-  id: Id
+  transport_address: TransportAddress
+  external_id?: string
+  attributes?: Dictionary<string, string>
+  roles?: string[]
+  version?: VersionString
+  min_index_version?: integer
+  max_index_version?: integer
```

Comment:
I believe all of those attributes are not optional. They can be nullable in the constructor, sometimes, but in the class itself and the XContent representation, they're always defined. For example, name is set to the empty string instead of null:

https://github.com/elastic/elasticsearch/blob/045f6a31f994f51d87a217be60251e060132c8a1/server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java#L242-L246

Reply:
ah you're right this is a refuse from when I still wanted to keep a single DiscoveryNode class, fixing this